### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by othalla**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryproject.org/en/master/contributing.html).

## Description

Small PR to clean some code.

### CODE REVIEW

File: celery/backends/amqp.py [hunk 1]
```diff
@@ -141,12 +141,11 @@ def wait_for(self, task_id, timeout=None, cache=True,
         if cache and cached_meta and \
                 cached_meta['status'] in READY_STATES:
             return cached_meta
-        else:
-            try:
-                return self.consume(task_id, timeout=timeout, no_ack=no_ack,
-                                    on_interval=on_interval)
-            except socket.timeout:
-                raise TimeoutError('The operation timed out.')
+        try:
```
Comment (thedrow) [inline celery/backends/amqp.py]: Why remove the else?

**Comment by othalla** [comment]
Thank you.
FYI, plan to do lots of PR like this one, and refactor if i can :)

Just one question, is there any reason not to have pylint in CI?

**Comment by thedrow** [comment]
We have flake8 which runs pylint.

**Comment by othalla** [comment]
They're quite different. Basically, they both respect pep8. Pylint reports more things, sometimes too strict but it can be configured.

**Pull request merged**